week4-ts/src/components/ResultSection.tsx

```diff
@@ -0,0 +1,31 @@
+import { Store } from "../types";
+import Skeleton from "../components/Skeleton";
+import StoreCard from "./StoreCard";
+
+interface ResultSectionProps {
+  storeInfo: Store[];
+  isCheck: boolean;
+  isLoading: boolean;
+}
+
+export default function ResultSection({
+  storeInfo,
+  isCheck,
+  isLoading,
+}: ResultSectionProps) {
+  return (
+    <>
+      {isLoading &&
+        new Array(10).fill(1).map((_, idx) => {
+          return <Skeleton key={idx} />;
+        })}
+      {!isLoading && storeInfo.length === 0 ? (
+        <div style={{ color: "#fff" }}>검색 결과가 없습니다.</div>
+      ) : (
+        storeInfo?.map((store: Store, idx: number) => (
+          <StoreCard key={idx} store={store} isCheck={isCheck} />
+        ))
+      )}
```

Comment:
```suggestion
  if(isLoading) {
    new Array(10).fill(1).map((_, idx) => {
      return <Skeleton key={idx} />;
    })
  } else {
    return (
      <>
        {storeInfo.length === 0 ? (
          <div style={{ color: "#fff" }}>검색 결과가 없습니다.</div>
        ) : (
          storeInfo?.map((store: Store, idx: number) => (
            <StoreCard key={idx} store={store} isCheck={isCheck} />
          ))
        )}
      </>
    )
  }
```
혹은 이런 식으로 조건부 렌더링을 하는 방식도 있을 것 같아여 !

Reply:
혹은 나중에 서버사이드 데이터 처리 상태를 로딩 상태 뿐만 아니라 에러 상태까지 확장시킨다면 이런 식으로 switch-case문을 이용한 조건부렌더링을 할 수도 있을 것 같아요 ! 

```javascript
function Result({ isEmpty, keywordLocationData, status }: ResultProps) {
  switch (status) {
    case 'LOADING':
      return <Loading />;
    case 'ERROR':
      return <ErrorFallback />;
    default:
      return isEmpty ? <NoResult /> : <CardList keywordLocationData={keywordLocationData} />;
  }
}
```

근데 사실 이거 다 갠취인 영역이라 정답은 없어서 갠취대로 갈기면 될 거 같긴 해여 ㅋ ㅋ

Reply:
아 저렇게 하면 되는구나 ㅋㅋ저런식으로 갈겨볼게여
switch문 상태 처리 코드도 참고할게요 감삼닷!